docs/sql-programming-guide.md

```diff
@@ -580,6 +580,9 @@ default local Hive metastore (using Derby) for you. Unlike the `createOrReplaceT
 Hive metastore. Persistent tables will still exist even after your Spark program has restarted, as
 long as you maintain your connection to the same metastore. A DataFrame for a persistent table can
 be created by calling the `table` method on a `SparkSession` with the name of the table.
+Notice that for `DataFrames` is built on Hive table, `insertInto` should be used instead of `saveAsTable`.
```

Comment:
Let us get rid of `Notice that for DataFrames is built on Hive table,`.  `insertInto` can work for any existing table. More importantly, `DataFrames` might be created from scratch.

Reply:
@gatorsmile Could you elaborate on your comment? The purpose of this sentence was to warn user to use `insertInto` when they are dealing DataFrames that created from Hive table. Since due to https://issues.apache.org/jira/browse/SPARK-16803, `saveAsTable` will not work on that special case. Or do you have any suggestions to make it more clear?

Reply:
This limitation is lifted in Spark 2.2. See https://issues.apache.org/jira/browse/SPARK-19152

Reply:
Ah, I see. In that case, I don't see the necessary to mention `insertInto` in this doc any more. I am closing this PR. Feel free to reopen it if necessary. 
I would also suggest to mark https://issues.apache.org/jira/browse/SPARK-22808 as reolvsed and linked it to https://issues.apache.org/jira/browse/SPARK-19152

Reply:
Done. Thanks!